Tests/SwiftJavaTests/JavaRepositoryTests.swift

```diff
@@ -0,0 +1,313 @@
+    configuration.packageToDownload = #""org.andrejs:json:1.2""#
+    configuration.remoteRepo = "https://jitpack.io"
+    // use local repo, since includeGroups only applied to mavenLocal
+    configuration.preferredLocalRepo = FileManager.default.homeDirectoryForCurrentUser.appendingPathComponent(".m2/repository").path
+
+    configuration.repositories = [.mavenLocal(includeGroups: ["commons-io"])]
+    return configuration
+  }()
+
+  static let jitpackJsonUsingCentralRepository: Configuration = {
+    var configuration = Configuration()
+    configuration.swiftModule = "JavaJson"
+    configuration.dependencies = [
+      JavaDependencyDescriptor(groupID: "org.andrejs", artifactID: "json", version: "1.2")
+    ]
+    return configuration
+  }()
+
+  static let andriodCoreInCentral: Configuration = {
+    var configuration = Configuration()
+    configuration.swiftModule = "JavaAndroidCommon"
+    configuration.dependencies = [
+      JavaDependencyDescriptor(groupID: "android.arch.core", artifactID: "common", version: "1.1.1")
+    ]
+    return configuration
+  }()
+
+  static let androidLifecycleInRepo: Configuration = {
+    var configuration = Configuration()
+    configuration.swiftModule = "JavaAndroidLifecycle"
+    configuration.dependencies = [
+      JavaDependencyDescriptor(groupID: "android.arch.lifecycle", artifactID: "common", version: "1.1.1")
+    ]
+    // using the following property to download to local repo
+    configuration.packageToDownload = #""android.arch.lifecycle:common:1.1.1""#
+    configuration.remoteRepo = "https://maven.google.com"
+    configuration.splitPackage = true
+
+    configuration.repositories = [
+      .maven(url: JavaRepositoryTests.localJarRepo/*, artifactUrls: [
+        JavaRepositoryTests.localPomRepo
+      ]*/)
+    ]
+    return configuration
+  }()
+}
+
+// MARK: - Download to local repo
+
+private extension SwiftJavaConfigurationShared.Configuration {
+  /// in json format, which means string needs to be quoted
+  var packageToDownload: String? {
+    get { javaPackage }
+    set { javaPackage = newValue }
+  }
+
+  var remoteRepo: String? {
+    get { outputJavaDirectory }
+    set { outputJavaDirectory = newValue }
+  }
+
+  /// whether to download jar and pom files separately
+  var splitPackage: Bool? {
+    get { writeEmptyFiles }
+    set { writeEmptyFiles = newValue }
+  }
+
+  var preferredLocalRepo: String? {
+    get { classpath }
+    set { classpath = newValue }
+  }
+
+  func downloadIfNeeded() async throws {
+    guard
+      let data = packageToDownload?.data(using: .utf8),
+      let descriptor = try? JSONDecoder().decode(JavaDependencyDescriptor.self, from: data),
+      let repo = remoteRepo
+    else {
+      return
+    }
+    let splitPackage = splitPackage ?? false
+
+    let process = Process()
+    process.executableURL = URL(fileURLWithPath: "/usr/bin/env")
+    process.arguments = [
+      "mvn", "dependency:get",
```

Comment:
we don't have maven on CI nodes so this will fail in CI; is there a way to test this functionality using invoking `swift-java resolve` instead? We don't need to be testing external tools and the `resolve` should cause a download after all